packages/a11y-testing/src/validators/validate.ts

```diff
@@ -0,0 +1,62 @@
+import { Rule, TestFacade } from '../types';
+
+import { SlotRule } from './rules';
+
+export const validateSlot = (rule: SlotRule, baseTestFacade: TestFacade) => {
+  const slot = rule.getData();
+  const slotProps = slot.props || [{}];
+  slotProps.map(props => {
+    const testFacade = baseTestFacade.forProps(props);
+
+    if (slot.checkClick) {
+      testFacade.afterClick(slot.name);
+    }
+
+    if (slot.checkSpaceKeyPressed) {
+      testFacade.pressSpaceKey(slot.name);
+    }
+
+    if (slot.checkEnterKeyPressed) {
+      testFacade.pressEnterKey(slot.name);
+    }
+
+    if (slot.wasOnclickExecuted) {
```

Comment:
For consistensy, can we rename this to checkOnClickWasExecuted, or something similar

Reply:
renamed :)